app.js

```diff
@@ -155,7 +164,22 @@ function drainAndShutdown(signal) {
         { signal },
         `received interrupt, starting drain over ${shutdownDrainTimeWindow} mins`
       )
-      DrainManager.startDrainTimeWindow(io, shutdownDrainTimeWindow)
+      DrainManager.startDrainTimeWindow(io, shutdownDrainTimeWindow, () => {
+        setTimeout(() => {
+          const staleClients = io.sockets.clients()
+          if (staleClients.length !== 0) {
+            logger.warn(
+              { staleClients: staleClients.map((client) => client.id) },
+              'forcefully disconnecting stale clients'
+            )
+            staleClients.forEach((client) => {
+              client.disconnect()
```

Comment:
Would this code make more sense in the drain manager?

Reply:
I would like to keep the DrainManager as is -- the DrainManager is more-or-less stateless and the drain process can get stopped/adjusted at any time, whereas there is no way back with the newly proposed marking as unhealthy at the end. Implementing a ShutdownManager with all the shutdown logic may be a something for a future task?